examples/mem-alloc-test/src/main.rs

```diff
@@ -166,22 +167,29 @@ fn worker_thread(
         let start_time = Instant::now();
 
         /* Create an array of x GB where x is a random number between 1 to 4 */
-        let mut large_vector = Vec::with_capacity(size);
-        large_vector.resize(size, 0);
+
+        // Create a layout based on the size and alignment
+        let layout = Layout::from_size_align(size, ALIGN).unwrap();
```

Comment:
Always aligning to the page size is stressful in some ways, but not very stressful in other ways, as we'll never end up with two allocations on the same page. It's probably better to pick the alignment randomly as well (with the restriction that the alignment must be a power of 2), and then check that the returned pointer has the correct alignment.